client/src/views/record/list.js

```diff
@@ -505,15 +510,31 @@ function (Dep, MassActionHelper, ExportHelper) {
              * @this module:views/record/list.Class
              */
             'click .record-checkbox': function (e) {
-                let $target = $(e.currentTarget);
+                const $target = $(e.currentTarget);
+
+                if (!this._lastChecked) {
```

Comment:
This if-block cause an issue when checked the first time that the item is not actually selected. Seems it's added by a mistake and should be removed. Then the following if to be changed to

```
if (e.shiftKey && this._lastChecked) {
```

Reply:
Good point, missed that.